worker-api/src/main/java/com/github/workerframework/api/TrackingMessageCreator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2015-2025 Open Text.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.workerframework.api;
+
+import java.util.Map;
+
+public interface TrackingMessageCreator {
+
+    TaskMessage createResponseTaskMessage(
+        final TaskMessage taskMessage,
+        final WorkerResponse response,
+        final Map<String, byte[]> responseContext,
+        final TrackingInfo trackingInfo,
+        final WorkerConfiguration workerConfiguration
+    );
+
+    TaskMessage createReportUpdateMessage(
+        final String correlationId,
+        final byte[] reportUpdatesTaskData,
+        final String routingKey
+    );
+
+    TaskMessage createInvalidTaskMessage(
+        final TaskMessage taskMessage,
+        final String message,
+        final String routingKey,
+        final WorkerConfiguration workerConfiguration
+    );
+}
```

Comment:
createInvalidTaskMessage common to WorkertaskImpl and WorkerQueueConsumerImpl, seemed right to pull out other two message creation methods.